@aapeliv's issue -- Let's work together!

Hi @aapeliv,

Thanks for giving this tool a whirl! I saw you'd starred. But I also noticed you'd pushed a commit to a fork, presumably to work around an issue you were running into.

Any chance we could work together to resolve in the main repo whatever problem you were running into? Would love to understand and fix what went wrong both for you and for others in the future who might run into it.

I'd love to at least hear what your setup is, and what went wrong. Bonus points if you'd be willing to help reproduce the issue or get us to a fix!

Thanks!
Chris

Here's the failing output:

```sh
$ bazel run //:refresh_compile_commands
INFO: Analyzed target //:refresh_compile_commands (5 packages loaded, 7 targets configured).
INFO: Found 1 target...
Target //:refresh_compile_commands up-to-date:
  bazel-bin/refresh_compile_commands.sh
INFO: Elapsed time: 0.431s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
 >>> Analyzing commands used in //cc/... 
Traceback (most recent call last):
  File "/home/aapeli/.cache/bazel/_bazel_aapeli/e96571fd1c49704bf5db6143f7865193/execroot/__main__/bazel-out/k8-fastbuild/bin/external/hedron_compile_commands/extract.runfiles/hedron_compile_commands/extract.py", line 208, in <module>
    for files, command in outputs:
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 619, in result_iterator
    yield fs.pop().result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/aapeli/.cache/bazel/_bazel_aapeli/e96571fd1c49704bf5db6143f7865193/execroot/__main__/bazel-out/k8-fastbuild/bin/external/hedron_compile_commands/extract.runfiles/hedron_compile_commands/extract.py", line 189, in _get_cpp_command_for_files
    files = _get_files(args)
  File "/home/aapeli/.cache/bazel/_bazel_aapeli/e96571fd1c49704bf5db6143f7865193/execroot/__main__/bazel-out/k8-fastbuild/bin/external/hedron_compile_commands/extract.runfiles/hedron_compile_commands/extract.py", line 81, in _get_files
    assert len(source_files) > 0, f"No sources detected in {compile_args}"
AssertionError: No sources detected in ['/usr/bin/clang', '-U_FORTIFY_SOURCE', '-fstack-protector', '-Wall', '-Wthread-safety', '-Wself-assign', '-fcolor-diagnostics', '-fno-omit-frame-pointer', '-MD', '-MF', 'bazel-out/k8-fastbuild/bin/external/boringssl/_objs/crypto/chacha-x86_64.pic.d', '-fPIC', '-iquote', 'external/boringssl', '-iquote', 'bazel-out/k8-fastbuild/bin/external/boringssl', '-isystem', 'external/boringssl/src/include', '-isystem', 'bazel-out/k8-fastbuild/bin/external/boringssl/src/include', '-Wa,--noexecstack', '-D_XOPEN_SOURCE=700', '-Wall', '-Werror', '-Wformat=2', '-Wsign-compare', '-Wmissing-field-initializers', '-Wwrite-strings', '-Wshadow', '-fno-common', '-std=c11', '-Wmissing-prototypes', '-Wold-style-definition', '-Wstrict-prototypes', '-no-canonical-prefixes', '-Wno-builtin-macro-redefined', '-D__DATE__="redacted"', '-D__TIMESTAMP__="redacted"', '-D__TIME__="redacted"', '-c', 'external/boringssl/linux-x86_64/crypto/chacha/chacha-x86_64.S', '-o', 'bazel-out/k8-fastbuild/bin/external/boringssl/_objs/crypto/chacha-x86_64.pic.o']
 >>> Finished extracting commands for //cc/... 
```

I suppose that's because this is some machine-optimized/dependent assembly for encryption so there's no C/C++ source. I think the dependency might be gRPC (I can isolate this and make a minimal failing WORKSPACE as well).

--- a/extract.py
+++ b/extract.py
@@ -75,7 +75,7 @@ def _get_headers(compile_args: List[str], source_path_for_sanity_check: Optional
 def _get_files(compile_args: List[str]):
     """Gets all source files clangd should be told the command applies to."""
     # Whole C-language family.
-    source_extensions = ('.c', '.cc', '.cpp', '.cxx', '.c++', '.C', '.m', '.mm', '.M')
+    source_extensions = ('.c', '.cc', '.cpp', '.cxx', '.c++', '.C', '.m', '.mm', '.M', '.S')
     source_files = [arg for arg in compile_args if arg.endswith(source_extensions)]
 
     assert len(source_files) > 0, f"No sources detected in {compile_args}"

